ext/opcache/zend_persist.c

```diff
@@ -87,6 +88,49 @@ static void zend_persist_op_array(zval *zv);
 static const uint32_t uninitialized_bucket[-HT_MIN_MASK] =
 	{HT_INVALID_IDX, HT_INVALID_IDX};
 
+static void mem_checksum_skip_list_add(void *p, uint32_t checked_area_size, uint32_t repetitions)
+{
+	ZEND_ASSERT(ZCG(mem_checksum_skip_list) != NULL);
+	char *base_ptr = (char *) ZCG(current_persistent_script)->mem + sizeof(zend_persistent_script);
+	ZEND_ASSERT((char *) p >= base_ptr);
+#if ZEND_DEBUG
+	ZEND_ASSERT(ZCG(mem_checksum_skip_list_count) < ZCG(mem_checksum_skip_list_capacity));
+#endif
+	zend_accel_skip_list_entry *entry = &ZCG(mem_checksum_skip_list)[ZCG(mem_checksum_skip_list_count)++];
+	entry->offset = (char *) p - base_ptr;
+	entry->checked_area_size = checked_area_size - sizeof(char*);
+	entry->repetitions = repetitions;
+}
+
+static int mem_checksum_skip_list_compare(const zend_accel_skip_list_entry *l, const zend_accel_skip_list_entry *r)
+{
+	if (l->offset < r->offset) {
+		return -1;
+	} else if (l->offset == r->offset) {
+		return 0;
+	} else {
+		return 1;
+	}
+}
+
+static void mem_checksum_skip_list_swap(zend_accel_skip_list_entry *l, zend_accel_skip_list_entry *r)
+{
+	zend_accel_skip_list_entry tmp = *r;
+	*r = *l;
+	*l = tmp;
+}
+
+static zend_accel_skip_list_entry *mem_checksum_skip_list_persist(void)
+{
+	zend_sort(ZCG(mem_checksum_skip_list), ZCG(mem_checksum_skip_list_count), sizeof(zend_accel_skip_list_entry), (compare_func_t) mem_checksum_skip_list_compare, (swap_func_t) mem_checksum_skip_list_swap);
```

Comment:
I think we don't need sorting? ``zend_persist.c`` stores data in a preallocated linear space, so the initial skip-list should be already sorted. I may be wrong. Please check this.